### PR TITLE
Support command-line argument `--option=value`

### PR DESCRIPTION
**Pull request by pensivej**
This PR relates to #2173 
This supports command-line argument `--option=value` syntax (not just `--option value`).
It is designed to replace the first occurence of "=" to " ".

### CODE REVIEW

**Comment by eggplantzzz** [comment]
Hey @pensivej, thanks for submitting this PR! However, there are a couple of issues here:
  - in calculating the `inputOptions` array, we should only put in the flagged command. In other words, if the user types `--option=true`, then we should only put `--option` into that array.
  - the `validOptions` as far as I can tell is currently being calculated correctly - I don't believe we will ever need to have `=` removed from it.
  - when we warn we also should only warn about the flagged option (don't include the value) - this means the output should look like 
```
> Warning: possible unsupported (undocumented in help) command line option: --option
```
and not 
```
> Warning: possible unsupported (undocumented in help) command line option: --option=true
```
Let me know if you have time to make these corrections!

**Comment by pensivej** [comment]
Sure, I will take a deeper look into it.

File: packages/truffle-core/lib/console.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Console extends EventEmitter {
     this.repl.on("exit", () => this.emit("exit"));
   }
 
-  start(callback) {
+  async start(callback) {
```
Comment (gnidan) [inline packages/truffle-core/lib/console.js]: Heads up: it's bad practice to make a `callback`-style function also `async`. Doing this has broken things in strange/unexpected ways.

Instead, I think you have to wrap the call to `this.repl.start` inside a `this.web3.eth.getAccounts().then(...)`.

**Comment by pensivej** [comment]
Oops wrong commits
I will make another pull request

**Comment by gnidan** [comment]
Heh so much for my quick review! Well, I'll copy my comment about the `async`+`callback` problem over to the new PR.

**Comment by pensivej** [comment]
@gnidan wow that was quite a quick review! I'll submit new PRs :)

**Pull request closed without merging**